training/tests/integration/conftest.py

```diff
@@ -232,14 +235,29 @@ def gnn_config(testing_modifications_with_temp_dir: DictConfig, get_tmp_paths: G
 
 
 @pytest.fixture
-def gnn_config_with_checkpoint(
-    gnn_config: tuple[DictConfig, str],
+def architecture_config_with_checkpoint(
+    request: pytest.FixtureRequest,
+    architecture_config: tuple[DictConfig, str],
     get_test_data: GetTestData,
 ) -> tuple[DictConfig, str]:
-    cfg, dataset_url = gnn_config
-    existing_ckpt = get_test_data(
-        "anemoi-integration-tests/training/checkpoints/testing-checkpoint-global-2025-07-31.ckpt",
-    )
+    cfg, dataset_url = architecture_config
+
+    model_type = request._pyfuncitem.callspec.params["architecture_config"][0]
+
+    # Pick checkpoint path based on the model
+    if "gnn" in model_type:
+        existing_ckpt = get_test_data(
+            "anemoi-integration-tests/training/checkpoints/testing-checkpoint-gnn-global-2025-07-31.ckpt",
```

Comment:
Also note that for now, this is a copy of the existing ckpt in the bucket with the name updated to account for the different architectures and still not break the current tests. Once we are happy with the changes here I can take care of cleaning the checkpoints bucket folder